### PR TITLE
fix(chart): Fix bugs ‘Older Dates with Year 1937 or less in charts is showing as NAN’

### LINKED ISSUE #24561: Older Dates with Year 1937 or less in charts is showing as NAN.
Hello, I am using superset version 2.1.0

For older dates around year 1937 and back I am not able to see correct dates in charts, it is showing as NAN.
Although in dataset it is being shown correctly.

Database I have used Databricks SQL, SQL server.

For example I have created a simple table chart by using this dummy date in dataset:

select  (current_date() - 31150 ) as dt1,  the chart shows NAN value

image

But if I run this select  (current_date() - 31105) as dt1, it shows correct date
image

So dates value 31150 days back from current date is not being shown correctly. But 31105 days back is correctly shown.
Can you please let me know if there is such limitation in charts and if it is so how can we rectify as we have dates older than 1930 year in our tables

### PR DESCRIPTION
**Pull request by wangkun723**
### SUMMARY

I made modifications in two places to try to solve the problem.The problem may be that the time data transmitted by the backend to the front-end is a timestamp, but when the length of a number timestamp number is greater than 12 (excluding symbols) when the time is less than 1937, the front-end JSON conversion will become 'BigNumber' and cannot be directly converted into formatted form.
### ADDITIONAL INFORMATION

- [x] Has associated issue:fixes #24561 
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by wangkun723** [comment]
How to obtain approval from maintenance personnel?

**Comment by sfirke** [comment]
I can't speak to the code but appreciate you working on this!

File: superset-frontend/plugins/plugin-chart-pivot-table/src/react-pivottable/TableRenderers.jsx [hunk 1]
```diff
@@ -423,7 +423,11 @@ export class TableRenderer extends React.Component {
           dateFormatters &&
           dateFormatters[attrName] &&
           typeof dateFormatters[attrName] === 'function'
-            ? dateFormatters[attrName](colKey[attrIdx])
+            ? dateFormatters[attrName](
```
Comment (justinpark) [inline superset-frontend/plugins/plugin-chart-pivot-table/src/react-pivottable/TableRenderers.jsx:426]: @wangkun723 it doesn't look safe to custom logic in each chart. The logic should be within dateFormatter. 
I also found [this PR](https://github.com/apache/superset/pull/23339) from @villebro which seems cover this case too.
Could you try with the above hotfix?

Comment (wangkun723) [reply superset-frontend/plugins/plugin-chart-pivot-table/src/react-pivottable/TableRenderers.jsx:426]: Okay, thank you for your guidance

**Comment by sfirke** [comment]
I can't replicate https://github.com/apache/superset/issues/24561 on 3.0.0rc2, following the same steps I use to reproduce the behavior on 2.1.0.  Maybe closed by the PR that @justinpark flagged as related, https://github.com/apache/superset/pull/23339 ?  (though #24561 wasn't Trino-specific).

@wangkun723 are you able to try out 3.0.0rc2 release candidate version or the `latest` version from Github and see if the original bug has already been fixed?  If so then this PR could probably be closed.

**Comment by wangkun723** [comment]
> 我无法按照我在 2.1.0 上重现行为的相同步骤在 3.0.0rc2 上复制[#24561 。](https://github.com/apache/superset/issues/24561)也许被 PR 关闭了@justinpark标记为相关，#23339？（尽管#24561不是 Trino 特有的）。
> 
> @wangkun723您是否可以尝试 3.0.0rc2 候选版本或`latest`来自 Github 的版本，看看原始错误是否已修复？如果是这样，那么这个 PR 可能会被关闭。

image
It still exists

**Comment by rusackas** [comment]
It seems that the issue this PR was attempting to solve is no longer an issue in current versions. That's probably how this slipped under the radar for so long... it's just not needed nowadays. If anyone is still experience the bug or sees a need for this PR, just say the word, so we can reopen it and see if it passes current CI. Thanks for the contribution!

**Pull request closed without merging**